Dockerfile.okd

```diff
@@ -0,0 +1,16 @@
+FROM registry.svc.ci.openshift.org/ocp/builder:rhel-8-base-openshift AS builder
```

Comment:
Shouldn't this be Fedora?

Reply:
@dtantsur we use the same builder images that are used in OCP for OKD today.

(The community has requested freely distributable builder imgages, but we don't have them yet.)

Reply:
Then why is ironic-image using Fedora? https://github.com/openshift/ironic-image/blob/master/Dockerfile.okd

What prevents you from using Fedora or CentOS here? RDO distributes IPA images for CentOS just as well.

Reply:
@dtantsur the ironic-image uses Fedora so that the EFI image is built using Fedora sources/packages. Arguably, CentOS Stream could be used, too, but since OKD is based on Fedora CoreOS, using Fedora seemed best to align things.

Ideally, the IPA resources would also be Fedora-based, but there are Fedora-based no builds afaict.

Reply:
Maybe use Fedora but download images from https://images.rdoproject.org/centos8/master/rdo_trunk/current-tripleo/? They themselves are CentOS based, but that should not matter.

Reply:
Is there a specific reason why it's important that Fedora is used here? Unless there is one (like with the ironic-image), I'd rather not diverge from how we do things everywhere else. cc @vrutkovs 

For the latter part, what's the right repo to pull the images from? Currently this'll use `https://images.rdoproject.org/stein/rdo_trunk/current-tripleo-rdo` (see https://github.com/openshift/ironic-ipa-downloader/pull/56/files#diff-4d7ed882fb9025e86dd94cc0b3034dec23a7201db1a0cf08109490a53528718fR10)

Reply:
@LorbusChris about the Fedora part, you will end up having huge inconsistencies between images. Ideally you want to have the same base image everywhere, you said it yourself that OKD is based on Fedora CoreOS, so using Fedora looks like the best choice here (as it is in ironic-image), or since you already diverged using Fedora in ironic-image, change that back to registry.svc.ci.openshift.org/ocp/builder:rhel-8-base-openshift
BTW the correct base image in the openshift registry is registry.ci.openshift.org/ocp/builder:rhel-8-base-openshift-4.9, the registry.svc.ci.openshift.org link is deprecated and we're using release-specific images since a while now

Reply:
It's important to note that Fedora is not used a base for payload/artifact images in OKD. Those are mostly UBI images.

We have to use Fedora (or CentOS) to build the ESP partition image because we copy EFI binaries (https://github.com/openshift/ironic-image/blob/master/Dockerfile.okd#L12-L13). We can't distribute RHEL bins like that in OKD.
Also note that the `ironic-image` that is eventually distributed as part of the payload is CentOS based - some of the RPM deps are not available in UBI and Fedora.

If you'd like to use Fedora as a builder here that's also fine and I'll make the change.

Reply:
Well, I assumed that an upstream distribution of OpenShift should be build-able without being a Red Hat OpenShift employee. Can this image be accessed without registration?

Reply:
>Can this image be accessed without registration?

No, it has RHEL's golang compiler. We can create temporary instructions on how to build one from Fedora/CentOS image though

Reply:
I see getting a public builder base as a separate topic which we can tackle later. For now, I'd just do things here as we do them everywhere in OKD in order to get us moving on this sooner.

Reply:
Okay, I see. I guess my impression of OKD was wrong.